src/writer/e2bdataanalysis.py

```diff
@@ -0,0 +1,258 @@
+        """Install Python package in sandbox"""
+        try:
+            result = await self.session.install_python_package(package_name)
+            return {
+                "package": package_name,
+                "status": "installed",
+                "output": result
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Package installation error: {str(e)}")
+
+    async def _upload_file(self, file_path: str, content: str) -> Dict[str, Any]:
+        """Upload file to sandbox"""
+        try:
+            await self.session.upload_file(file_path, content)
+            return {
+                "status": "uploaded",
+                "file_path": file_path
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"File upload error: {str(e)}")
+
+    async def _download_file(self, file_path: str) -> Dict[str, Any]:
+        """Download file from sandbox"""
+        try:
+            content = await self.session.download_file(file_path)
+            return {
+                "content": content,
+                "file_path": file_path
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"File download error: {str(e)}")
+
+    async def _list_files(self, path: str = ".") -> Dict[str, Any]:
+        """List files in sandbox directory"""
+        try:
+            files = await self.session.list_files(path)
+            return {
+                "files": files
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"File listing error: {str(e)}")
+
+    async def _get_environment_info(self) -> Dict[str, Any]:
+        """Get information about the sandbox environment"""
+        try:
+            python_version = await self._execute_code("import sys; print(sys.version)")
+            installed_packages = await self._execute_code("!pip list")
+            
+            return {
+                "python_version": python_version["output"],
+                "installed_packages": installed_packages["output"]
+            }
+        except Exception as e:
+            self.outcome = "error"
+            raise RuntimeError(f"Environment info error: {str(e)}")
+
+    async def run(self):
+        try:
+            # Get required fields
+            api_key = self._get_field("api_key")
+            operation = self._get_field("operation")
+
+            # Initialize session
+            await self._initialize_session(api_key)
+
+            # Get optional fields
+            code = self._get_field("code", True)
+            package_name = self._get_field("package_name", True)
+            file_path = self._get_field("file_path", True)
+            content = self._get_field("content", True)
+            timeout = int(self._get_field("timeout", True, "30"))
```

Comment:
Some fields are not JSON
```suggestion
            code = self._get_field("code")
            package_name = self._get_field("package_name")
            file_path = self._get_field("file_path")
            content = self._get_field("content")
            timeout = int(self._get_field("timeout", False, "30"))
```